vitest.shared.mts

```diff
@@ -7,6 +7,14 @@ import { playwright } from '@vitest/browser-playwright';
 const CURRENT_DIR = dirname(fileURLToPath(import.meta.url));
 const WORKSPACE_ROOT = resolve(CURRENT_DIR, './');
 
+/**
+ * Packages referenced in tests that run both in Node.js and the browser.
+ * The tests are excluded in the browser at runtime using conditional skips,
+ * but Vitest still tries to bundle them, which can lead to issues.
+ * To avoid this, we externalize them from the browser build.
+ */
+const EXTERNALIZED_PACKAGES = ['execa'];
```

Comment:
This PR uncovered another problem: tests that are skipped with `it.skipIf(!isJSDOM)`, they're being bundled in the browser build without ever being executed, this causes unnecessary overhead for the build, and coverage. On top of that loading Node.js only dependencies in such builds causes unnecessary friction.

Reply:
When dealing with the vitest migration, I thought about having 

- `MyTest.test.ts` - any runner
- `MyTest.test.browser.ts` - only browser
- `MyTest.test.jsdom.ts` - only jsdom

Though I was doing too many changes already, so I dropped it.